api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java

```diff
@@ -0,0 +1,74 @@
+package org.pmiops.workbench.interceptors;
+
+import com.google.api.client.http.HttpResponseException;
+import com.google.api.services.oauth2.model.Userinfoplus;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.pmiops.workbench.auth.UserInfoService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpMethod;
+import org.springframework.stereotype.Service;
+import org.springframework.web.servlet.handler.HandlerInterceptorAdapter;
+
+/**
+ * Intercepts all non-OPTIONS API requests to ensure they have an appropriate auth token.
+ */
+@Service
+public class AuthInterceptor extends HandlerInterceptorAdapter {
```

Comment:
Possibly, but when you set up your interceptor you can specify what paths it applies to... so it amounts to much the same thing.

I found Spring Security documentation a little bit overwhelming and perhaps overkill for what we're trying to accomplish.